overlord/ifacestate/helpers.go

```diff
@@ -376,27 +376,21 @@ ConnsLoop:
 		staticPlugAttrs := connState.StaticPlugAttrs
 		staticSlotAttrs := connState.StaticSlotAttrs
 
-		// XXX: Refresh the copy of the static connection attributes for "content"
-		// and "system-files" interfaces.
-		// This is a partial and temporary solution to https://bugs.launchpad.net/snapd/+bug/1825883
-		// and https://bugs.launchpad.net/snapd/+bug/1942266.
-		switch plugInfo.Interface {
-		case "content":
-			var plugContent, slotContent string
-			plugInfo.Attr("content", &plugContent)
-			slotInfo.Attr("content", &slotContent)
-
-			if plugContent != "" && plugContent == slotContent {
+		iface, _ := interfaces.ByName(plugInfo.Interface)
+
+		if iface != nil {
+			updateStaticAttrsResult := iface.UpdateStaticAttrs(plugInfo, slotInfo)
```

Comment:
I have some vague recollection that a different approach we discussed to answer part of this question (the Can part if I understand things here) would be to rerun the autoconnection policy check for the pair of plug and slut? as essentially the methods we are adding are kind of repeating part of that in code and is a bit of duplication and things that need to be kept in sync

Reply:
hmm. I see what you are saying about re-expressing the autoconnection decl policy in code, although I think that it should actually be connection policy check not autoconnection policy check, as the update should happen for interfaces that were previously manually connected, even if they don't autoconnect (example would be content interface with matching `content` attrs but different publishers, where someone had manually run `snap connect ...` You still probably want static attrs to update.).

I also wonder if there still needs to be some check that figures out on an interface by interface basis whether or not there even are static attrs to be reloaded. This maybe would be worth benchmarking